orttraining/orttraining/python/training/_ortmodule_execution_manager.py

```diff
@@ -0,0 +1,268 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+from . import _utils, _ortmodule_utils, _ortmodule_output_transformation as _ortmodule_io
+
+from onnxruntime.capi.onnxruntime_inference_collection import OrtValue
+from onnxruntime.capi import _pybind_state as C
+
+import io
+import inspect
+import onnx
+import onnxruntime
+import torch
+from torch.utils.cpp_extension import ROCM_HOME
+
+ONNX_OPSET_VERSION = 12
+
+class ExecutionManager(object):
+    def __init__(self, module):
+        # Original and flattened (tranformed) output module
+        self._original_module = module._base_module
+        self._flattened_output_module = module
+
+        # Exported model
+        self._model = None
+        
+        # Model after inference optimization or gradient building.
+        self._optimized_model = None
+        self._module_graph_builder = None
+        self._graph_info = None
+
+        # TrainingAgent or InferenceAgent
+        self._execution_agent = None
+
+        # Debug flags
+        self._save_onnx = False
+        self._save_onnx_prefix = ''
+
+        self._is_training = None
```

Comment:
```suggestion
        self._export_mode = None
```
Or maybe "execution_mode" or "training_mode". 
Value can be either `torch.onnx.TrainingMode.TRAINING`or `torch.onnx.TrainingMode.EVAL`